dev/io.openliberty.data.internal_fat_validation/test-applications/DataValidationTestApp/src/test/jakarta/data/validation/web/DataValidationTestServlet.java

```diff
@@ -190,11 +217,35 @@ public void testSaveValidRecord() {
         assertEquals(Collections.EMPTY_SET, violations);
     }
 
+    /**
+     * Save a Java class (no entity annotation) without any constraint violations.
+     * Modify the entity in the database such that it violates constraints for Positive.
+     * Verify that the invalid entity can be updated to be made valid again.
+     */
+    @Test
+    public void testUpdateFixInvalidNegative_Class() {
+        creatures.save(new Creature(900l, "Moose", "Alces alces", //
+                        BigDecimal.valueOf(47513117l, 6), BigDecimal.valueOf(-91169845l, 6), //
+                        ZonedDateTime.of(2023, 8, 7, 15, 18, 19, 999, ZoneId.of("America/Chicago")).toOffsetDateTime(), //
+                        459.2f));
+
+        // Validation does not apply to updates made in ways that do not involve the class
+        assertEquals(true, creatures.updateByIdSetWeight(900L, -452.9f));
+
+        // Validation does not apply to find operations.
+        Creature c = creatures.findById(900l).orElseThrow();
```

Comment:
These tests seem fine to me, but I'm curious as to whether you think it would be worth anything (if only to someone reading through the test in future) to show 'c' failing validation at this point (like you do in the previous test) and then showing the new 'c' passing validation at the end of the method.

Reply:
Good idea. I added it.